Raven db 7873

talweiss1982 opened the pull request.

Commit:
RavenDB-7873
When escaping qouted query we should not escaped Lucene special chars
Fixing wrong test assertion in Kamran.cs
Adding a new test to check for the case where we have both whitespace and special Lucene chars in the same query

maximburyak committed:
RavenDB-7921 Output connectivity error in cluster UI

Commit:
Merge branch 'v3.5' of https://github.com/ravendb/ravendb into RavenDB-7873

Commit:
Merge branch 'RavenDB-7921' of https://github.com/maximburyak/ravendb into RavenDB-7873

Commit:
RavenDB-XXXX (issues is down) Raft logging
Added more logs around http transport and raft requests

ayende commented on Rachis/Rachis/Transport/HttpRaftRequest.cs:
```diff
@@ -54,8 +57,15 @@ private Task SendRequestInternal(Func<HttpRequestMessage> getRequestMessage)
+                }
+                catch (Exception e)
+                {
+                    _log.ErrorException($"Error while sending a request {requestMessage.Method} {requestMessage.RequestUri}\r\n content:{requestMessage.Content}\r\n{requestMessage.Headers}",e);
```
Will this actually give you the actual content?

ayende commented on Rachis/Rachis/Transport/HttpTransportSender.cs:
```diff
@@ -256,15 +263,20 @@ public void Send(NodeConnectionInfo dest, CanInstallSnapshotRequest req)
+                    }
+                    if (httpResponseMessage.StatusCode == HttpStatusCode.NotAcceptable)
+                    {
+                        _log.Warn("Error checking if can install snapshot to {0}. Status: NotAcceptable\r\nreason:{1}", dest.Name, httpResponseMessage.ReasonPhrase);
```
Add the reply content here as well

ayende commented on Rachis/Rachis/Transport/HttpTransportSender.cs:
```diff
@@ -275,22 +287,26 @@ public void Send(NodeConnectionInfo dest, CanInstallSnapshotRequest req)
+                        return;
+                    } else if (httpResponseMessage.StatusCode == HttpStatusCode.NotAcceptable)
+                    {
+                        _log.Warn("Error requesting vote from {0}. Status: NotAcceptable\r\nreason:{1}", dest.Name, httpResponseMessage.ReasonPhrase);
```
Add reply here too

ayende commented on Rachis/Rachis/Transport/HttpTransportSender.cs:
```diff
@@ -309,15 +325,20 @@ public void Send(NodeConnectionInfo dest, TimeoutNowRequest req)
+                    }
+                    if (httpResponseMessage.StatusCode == HttpStatusCode.NotAcceptable)
+                    {
+                        _log.Warn("Error sending timeout now to {0}. Status: NotAcceptable\r\nreason:{1}", dest.Name, httpResponseMessage.ReasonPhrase);
```
Add reply here too

ayende commented on Rachis/Rachis/Transport/HttpTransportSender.cs:
```diff
@@ -330,17 +351,23 @@ public void Send(NodeConnectionInfo dest, DisconnectedFromCluster req)
                     }
+                    if (httpResponseMessage.StatusCode == HttpStatusCode.NotAcceptable)
+                    {
+                        _log.Warn("Error sending disconnecton notification to {0}. Status: NotAcceptable\r\nreason:{1}", dest.Name, httpResponseMessage.ReasonPhrase);
```
Add reply here too

Commit:
Raven-????
Added support to stringify entries and snapshot content
Now re-run event loop if exception is thrown

talweiss1982 commented:
Added catch exception in event loop + logging
Added ToString to custom content
